### DISCUSSION TITLE
Ontime/SSB example datasets create table fields with `not null`

### DISCUSSION THREAD

**Discussion by sundy-li** [databendlabs/databend#4342] [General]
I think it's better to define the table schema as not null as possible.
This can show better performance in all queries.

https://github.com/datafuselabs/databend/blob/a54a4632f44c329d42ac71bfa7c0740034a00b8c/cli/src/cmds/generates/ddls/ontime.sql

cc @ZhiHanZ  @wubx

**Comment by Xuanwo** [comment]
I thought we are `not null` by default? Based on discussion over https://github.com/datafuselabs/databend/discussions/4234

**Reply by sundy-li** [reply]
Based on that discussion, it's null by default..

**Reply by Xuanwo** [reply]
Oh, must be my fault...

**Comment by wubx** [comment]
Can we use :
```
Year                            UInt16 not null default '',
```
replace 
```
Year                            UInt16,
```

**Reply by sundy-li** [reply]
`Year UInt16 not null` defaults to 0.
`C String not null` defaults to ''.

You can change the default expression, but please don't use `default ''` for `primitive` columns.
